extensions/amp-smartlinks/amp-smartlinks.md

```diff
@@ -0,0 +1,102 @@
+    <td>nodisplay</td>
+  </tr>
+</table>
+
+## Overview
+
+ At [Narrativ](https://narrativ.com/), we transform static commerce links into dynamic, multimerchant nodes. With a library of millions of products matched to expert reviews from top commerce publishers, we lift publisher revenue through real-time bidding and data solutions.
+
+This AMP extension is our Linkmate service in AMP. See the full documentation for Linkmate [here](http://docs.narrativ.com/en/stable/linkmate.html).
+
+## Getting started
+
+Your account must be a member of our Linkmate program to use this feature. For more information about this program, feel free to contact your account manager or [hello@narrativ.com](mailto:hello@narrativ.com).
+
+NOTE: If you plan to use `amp-smartlinks` alongside other affiliate partners you will need to specify the meta tag shown below. The tag will specify the order in which the affiliate tags fire.
+
+In your AMP page you will have to add the following snippets:
+
+```html
+<!doctype html>
+<html ⚡>
+<head>
+  ...
+  <meta name="amp-link-rewriter-priorities" content="amp-smartlinks amp-other-affiliate">
+  <script async custom-element="amp-smartlinks" src="https://cdn.ampproject.org/v0/amp-smartlinks-0.1.js"></script>
+  ...
+  <script async src="https://cdn.ampproject.org/v0.js"></script>
+</head>
+<body>
+    ...
+    <amp-smartlinks
+        layout="nodisplay"
+        nrtv-account-name="supercoolpublisher"
+        linkmate="true">
+    </amp-smartlinks>
+    ...
+</body>
+</html>
+```
+
+## Attributes
+
+<table>
+  <tr>
+    <td class="col-fourty"><strong>nrtv-account-name</strong></td>
+    <td><strong>Required</strong></td>
+    <td>Your Narrativ account name given to you by your account manager. Need to know your Narrativ account name? Log into <a href="https://dashboard.narrativ.com/#/login">dashboard.narrativ.com</a> and go to <a href="https://dashboard.narrativ.com/#/publisher/account/setup">setup</a> to see your account name in the snippet, or reach out to your account manager for support as needed.</td>
+  </tr>
+  <tr>
+    <td class="col-fourty"><strong>linkmate</strong></td>
+    <td><strong>Required</strong></td>
+    <td>Field to flag if you want Linkmate to run on an article. Expected values are <code>true</code> or <code>false</code>.</td>
+  </tr>
+  <tr>
+    <td class="col-fourty"><strong>exclusive-links</strong></td>
+    <td><strong>Optional</strong></td>
+    <td>Field to flag if you want exclusive links on an article. Expected values are <code>true</code> or <code>false</code>.</td>
+  </tr>
+  <tr>
+    <td class="col-fourty"><strong>link-attribute</strong></td>
+    <td><strong>Optional</strong></td>
+    <td>If you store the "plain" url for a link in a different element attribute than <code>href</code> you can specify so here. Default value: <code>href</code>.</td>
```

Comment:
I don't think this is really possible with AMP, since custom javascript can't be used. Does it make sense to remove this to streamline the API for your user?

Reply:
I think replacing different attribute other than `href` for anchor tag is possible, however I don't find replacing attribute other than `href` to be useful.  @PhilWinchester what is an example value for `link-attribute`?

Reply:
@zhouyx @Gregable sometimes we build on top of other providers' solutions.  In this case, the `href` attribute may have been modified server-side to include a 3rd-party link wrapper, and the original url would be found in an attribute like `data-original-url`.

Full example might be something like:
```
<a href="https://thirdparty.com/redirect?url=destination.com" data-original-url="https://destination.com">
```

In this case, we would want to pull the original url from the data-original-url attribute, to run through our own transformations, and finally insert into `href`.

Reply:
Understood, seems reasonable.